### PR TITLE
feat: default to TS for file extension and support ext flag in more scenarios

### LINKED ISSUE #11220: Default file extension inconsistency in the `deno run` cli
```
curl https://deno.land/std/examples/welcome.ts | deno run - 
```
Will run stdin as a TypeScript file.

https://github.com/denoland/deno/blob/0385a990529399775b76db4c27704faa4a092efa/cli/main.rs#L796

However,
```
deno run extensionless_file
```
execute `extensionless_file` as a JavaScript file.

I believe we should ideally consolidate the default behaviour of `deno run` when the file extension is not specified. Maybe for Deno 2.0?

Also Deno has different default in general regarding the `--ext` flag: `deno fmt` uses `ts` by default, while `deno eval` uses `js`.

### PR DESCRIPTION
**Pull request by crowlKats**
Closes #11220
Currently does lint, fmt, and repl

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
This looks good to me, but I want @dsherret to take a look as well before we merge

**Review by lucacasonato — COMMENTED** [review]
We should really lift this up into `deno_graph` after 2.0 is out the door...

File: cli/file_fetcher.rs
Comment (crowlKats) [reply cli/file_fetcher.rs:149]: no, as per what you have said in https://github.com/denoland/deno/issues/11220#issuecomment-2324359588
> extensionless files default to TS everywhere (entrypoint and imported)

**Review by bartlomieju — APPROVED** [review]
LGTM, let's land it and if Luca has concerns we can fix them tomorrow.

**Pull request merged**
